manifests/0000_70_cluster-network-operator_05_daemonset.yaml

```diff
@@ -0,0 +1,117 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: network-plugin-install
+  namespace: openshift-network-operator
+  annotations:
+    kubernetes.io/description: |
+      This daemonset installs all necessary CNI binaries on every host across the cluster
+spec:
+  selector:
+    matchLabels:
+      app: network-plugin-install
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: network-plugin-install
+        component: network
+        type: infra
+        openshift.io/component: network
+    spec:
+      hostNetwork: true
+      nodeSelector:
+        kubernetes.io/os: linux
+      priorityClassName: "system-node-critical"
+      tolerations:
+      - operator: Exists
+      initContainers:
+      - name: multus-bin-copy
+        image: "quay.io/openshift/origin-multus-cni:4.3"
+        command: ["/entrypoint/cnibincopy.sh"]
+        volumeMounts:
+        - mountPath: /entrypoint
+          name: cni-binary-copy
+        - mountPath: /host/opt/cni/bin
+          name: cnibin
+        - mountPath: /host/etc/os-release
+          name: os-release
+          readOnly: true
+        env:
+        - name: RHEL7_SOURCE_DIRECTORY
+          value: "/usr/src/multus-cni/rhel7/bin/"
+        - name: RHEL8_SOURCE_DIRECTORY
+          value: "/usr/src/multus-cni/rhel8/bin/"
+        - name: DEFAULT_SOURCE_DIRECTORY
+          value: "/usr/src/multus-cni/bin/"
+      - name: routeoverride-cni-bin-copy
+        image: "quay.io/openshift/origin-multus-route-override-cni:4.4"
+        command: ["/entrypoint/cnibincopy.sh"]
+        volumeMounts:
+        - mountPath: /entrypoint
+          name: cni-binary-copy
+        - mountPath: /host/opt/cni/bin
+          name: cnibin
+        - mountPath: /host/etc/os-release
+          name: os-release
+          readOnly: true
+        env:
+        - name: RHEL7_SOURCE_DIRECTORY
+          value: "/usr/src/route-override/rhel7/bin/"
+        - name: RHEL8_SOURCE_DIRECTORY
+          value: "/usr/src/whereabouts/rhel8/bin/"
+        - name: DEFAULT_SOURCE_DIRECTORY
+          value: "/usr/src/route-override/bin/"
+      - name: whereabouts-cni-bin-copy
+        image: "quay.io/openshift/origin-multus-whereabouts-ipam-cni:4.4"
+        command: ["/entrypoint/cnibincopy.sh"]
+        volumeMounts:
+        - mountPath: /entrypoint
+          name: cni-binary-copy
+        - mountPath: /host/opt/cni/bin
+          name: cnibin
+        - mountPath: /host/etc/os-release
+          name: os-release
+          readOnly: true
+        env:
+        - name: RHEL7_SOURCE_DIRECTORY
+          value: "/usr/src/whereabouts/rhel7/bin/"
+        - name: RHEL8_SOURCE_DIRECTORY
+          value: "/usr/src/whereabouts/rhel8/bin/"
+        - name: DEFAULT_SOURCE_DIRECTORY
+          value: "/usr/src/whereabouts/bin/"
+      - name: cni-plugins-bin-copy
+        image: "quay.io/openshift/origin-container-networking-plugins:4.3"
```

Comment:
@danwinship : this is specifically what I am wondering. What will happen here when we deploy the CNO?

What I've done is create a daemonset whos intention is to run before the network plugin. I am however unsure of what binds these `image` fields to whatever will be deployed at "runtime". It's obvious that these fields are overwritten at some point in CI, but I have no clue of when and how? Am I missing some configuration to make this work?

Reply:
ah, if you want to add a new image there are a bunch of steps you have to go through.

but we shouldn't need a new image here; the existing CNI plugins image already has what we need; you just need a different DaemonSet to run it in the case where multus is disabled, right?

Reply:
> but we shouldn't need a new image here; the existing CNI plugins image already has what we need

That's what I want to do, just re-use those images (which are already defined in the CNO template [here](https://github.com/openshift/cluster-network-operator)) but here too. 

Basically what I want is to have a daemonset that runs before the CNO starts deploying any network plugin. This daemonset just copies these binaries on to every host in the cluster. But using it, we can be certain that it is the only thing that performs this operation and equally importantly we only have one version and location of the script that perform this (which now is defined in `manifests/0000_70_cluster-network-operator_03_configmap.yaml`).

This idea is thus:

1) Deploy this daemonset
2) Deploy the CNO
3) CNO deploys the network plugin(s)

This order also ensures that the binaries are there when the network plugin starts and requires them to start setting up the network.

Reply:
ah, sorry, yes, as long as you refer to the image by exactly the same name as it appears in `manifests/image-references`, then CI and ART will do the appropriate substitutions when building release images.

Reply:
Yeah, this seems to have built and deployed fine.